dnscrypt-proxy/coldstart.go

```diff
@@ -122,11 +122,15 @@ func addColdStartListener(
 	listenAddrStr string,
 	captivePortalHandler *CaptivePortalHandler,
 ) error {
-	listenUDPAddr, err := net.ResolveUDPAddr("udp", listenAddrStr)
+	network := "udp"
+	if isDigit(listenAddrStr[0]) {
```

Comment:
This is a bit confusing. Maybe adding an `isIPv4()` function or using `! strings.HasPrefix(listenAddrStr, "[")` would be more self-explanatory.

Reply:
😂
explanatory Vs efficient

Reply:
Appended 8d43ce9 introducing `isIPv4` as a variable.